### PR TITLE
DF-714: Accessibility for date fields

### PR DESCRIPTION
**Pull request by TNA-Allan**
Ticket URL: https://national-archives.atlassian.net/browse/DF-714

**Please Do Not Merge, until FE is completed**

## About these changes

- Makes the individual date fields have the label with the input. - Day, Month, Year with GDS pattern.
- Added div as with [GDS](https://design-system.service.gov.uk/components/date-input/) pattern.
- FE stylings

## How to check these changes

- Navigate to Search - Records at the National Archives
- Observe Day, Month Year labels above the input fields

## Before assigning to reviewer, please make sure you have

- [x] Checked things thoroughly before handing over to reviewer.
- [x] Checked PR title starts with ticket number as per project conventions to help us keep track of changes.
- [x] Ensured that PR includes only commits relevant to the ticket.
- [x] Waited for all CI jobs to pass before requesting a review.
- [x] Added/updated tests and documentation where relevant.

## Merging PR guidance

Follow [docs\developer-guide\contributing.md](https://nationalarchives.github.io/ds-wagtail/developer-guide/contributing/)

## Deployment guidance

Follow [docs\infra\environments.md](https://nationalarchives.github.io/ds-wagtail/infra/environments/)

### CODE REVIEW

File: etna/core/widgets.py [hunk 1]
```diff
@@ -19,7 +19,10 @@ class TextInputWithLabel(LabelWidgetMixin, forms.TextInput):
     input to have its own label.
     """
 
-    pass
+    def get_context(self, name, value, attrs):
+        context = super().get_context(name, value, attrs)
+        context["widget"]["label"] = self.label
+        return context
```
[the remark this answers is not in the document]
Comment (TNA-Allan) [reply etna/core/widgets.py:25]: The [requirement](https://national-archives.atlassian.net/browse/DF-714?focusedCommentId=50577) wants these label as well in addition to parent date widget.

[the remark this answers is not in the document]
Comment (TNA-Allan) [reply etna/core/widgets.py:25]: @ababic are you fine with this BE change, I can then pass it over to FE ?

[the remark this answers is not in the document]
Comment (ababic) [reply etna/core/widgets.py:25]: I see now. This is fine!

**Review by ababic — APPROVED** [review]

File: templates/search/blocks/search_filters.html [hunk 2]
```diff
@@ -48,29 +48,23 @@ <h3 class="sr-only">Edit filters</h3>
                             <h4 class="search-filters__accordion-section-heading" id="record_covering_date">Dates</h4>
                         </legend>
 
+                        <div>{{ form.covering_date_from.label_tag }}</div>
```
Comment (ababic) [inline templates/search/blocks/search_filters.html:51]: I don't think `label_tag` is right here. What we really should have here is a `fieldset` tag wrapping the entire field, and the label value used to set the `legend` for that fieldset.

Comment (sdwilkes) [reply templates/search/blocks/search_filters.html:51]: Hi @JohnHeeryTNA , does this need to change before the PR can be reviewed? Thanks.

Comment (JohnHeeryTNA) [reply templates/search/blocks/search_filters.html:51]: @sdwilkes I've had a look at this and I think the rendered html is correct and following the GDS pattern: https://design-system.service.gov.uk/components/date-input/

According to that we should only have one <fieldset> which wraps the input fields, which is what we have. 

@ahosgood do you have a take on this?

Comment (ahosgood) [reply templates/search/blocks/search_filters.html]: I can't seem to see this in the list of changes (https://github.com/nationalarchives/ds-wagtail/pull/1074/files) but yes, there should be a single `<fieldset>` that encompasses all three date inputs as a set of fields and each `<input>` (day, month and year) should have a corresponding `<label>` element which describes it.

Comment (ahosgood) [reply templates/search/blocks/search_filters.html]: ...and a single `<legend>` for the `<fieldset>` which describes the set of fields as a whole

**Review by ahosgood — APPROVED** [review]

**Pull request merged**